CMakeLists.txt

```diff
@@ -169,7 +169,7 @@ adios_option(Catalyst   "Enable support for in situ visualization plugin using P
 adios_option(AWSSDK     "Enable support for S3 compatible storage using AWS SDK's S3 module" AUTO)
 include(${PROJECT_SOURCE_DIR}/cmake/DetectOptions.cmake)
 
-if(ADIOS2_HAVE_CUDA)
+if(ADIOS2_HAVE_CUDA OR ADIOS2_HAVE_Kokkos_CUDA)
```

Comment:
When we link to Kokkos the architecture should be set to whatever was used for Kokkos. We should not try to set another one.

Reply:
Kokkos does not set the architecture for the whole project but only to the cmake targets which `links` depends on it. the enable_language(CUDA) Will still need this variable.

The warnings are in current master/release_29 as shown here:

https://open.cdash.org/index.php?project=ADIOS&filtercount=1&showfilters=1&field1=revision&compare1=63&value1=a9ad351ea9e870ffade25ae2f76059a5b0d615a3

Reply:
You are right, I forgot we are treating the them as CUDA files. In this case, I think we should wait for this:
https://github.com/kokkos/kokkos/issues/5891

We do not want to risk setting another architecture than what was used for building Kokkos.

Reply:
I think that we should set for the whole adios2 project the same arch as we use for ADIOS2 CXX: C++11. For the particular components that use Kokkos we already set them to what kokkos use.

I am against having these warnings until kokkos team fixes this, if they do fix it.

Reply:
> For the particular components that use Kokkos we already set them to what kokkos use.

Hmm, We do this by manually setting it to C++17 but I am not sure if `kokkos_compilation()` or `target_links_library(kokkos::kokkos)` does set the CUDA_ARCHITECTURE automatically.

Reply:
Kokkos does seems to inject the -arch=XXYYZZ in the `INTERFACE_COMPILE_OPTIONS` `INTERFACE_LINK_OPTIONS` but the target for CUDA_ARCHITECTURE remains unset.

We need this to both set the default CMAKE_CUDA_ARCHITECTURE which is needed for the CI and also affect the cuda  example. And in the targets that use kokkos backend we need to  set it to whatever Kokkos give us.

Same thing for CMAKE_CUDA_STANDARD, however, kokkos already propagates this when you link your cmake target to kokkos::kokkos which for 3.7 seems to default to c++14.

Reply:
An also, we might want to use a different `CUDA_ARCHITECTURE` than kokkos, what if you building ADIOS2 with the kokkos RPM or DEB pkg. In that case the package would have been configured with a old cuda arch but you might want to use a new cuda arch.

Reply:
Let's do the following, leave it like this for the release and I will change it once Kokkos exposes the architecture. Acceptable?

Side note. If the Kokkos package was configured with an old cuda arch, our code will not work. I'll make a note in the documentation that users should set the `CUDA_ARCHITECTURE` to the same used for configuring Kokkos.

Reply:
> Let's do the following, leave it like this for the release and I will change it once Kokkos exposes the architecture. Acceptable?

Sounds good, if kokkos address it, we can comeback to this.